### DISCUSSION TITLE
Switched the update server to the Winter-API — what will happen next?

### DISCUSSION THREAD

**Discussion by nathanlesage** [wintercms/winter#166] [Q&A] [answered]
Hey y'all,

first of all, I am so very grateful that you continue the October timeline in an OSS manner, and thanks specifically to @LukeTowers for finding my tweet and pinging me that there's a fork.

I have now reverted my personal page back to October and according to the [instructions in one of the discussions](https://github.com/wintercms/winter/discussions/101) here set the update server URL to the Winter API.

Now I'm curious as to what will happen once the API is functional and the first Winter update will be pulled onto my October instance?

Specifically:

* I use a few of the rainlab-plugins and I've seen that you were successful in forking these as well – will new plugin updates at one point automatically come from the Winter-forks?
* Which files are actually NOT overwritten by the updater? (Just asking because I'm really angry at OCMS for closing the source and want to purge that name from all files)
* Is there anything I should keep out to watch which might break once my pages depart completely from the October branches?

Lastly, I found the update instructions for existing websites linked above — however on the page you have linked to in the migration guide on your website, a.k.a. [Luke's comment here](https://github.com/wintercms/winter/issues/5#issuecomment-797607263), this info is missing and it would be good if you could edit that post to add a section on how to migrate existing web installers just so that people like me which are not using the composer-installer have an easier time finding that single setting that enables us to get Winter updates; I think that would be good to have :)

Thanks y'all! Happy to continue contributing here!

P.S.: Prepare for an incomping sponsorship.

**Comment by LukeTowers** [comment, marked as answer]
The update & web installer delivery part of the marketplace API is being worked on, so as soon as that's ready a new update will become available through the backend. Once that happens, you will be able to update like it was a normal October update, and the updater will replace `bootstrap/autoload.php`, `modules/`, `vendor/` and add `composer.json` and `composer.lock`. This update will most likely be targeting the 1.0 branch (i.e. you'll be on Winter CMS 1.0, so you won't have access to the latest features) and then we'll introduce new functionality at some point to make it so that people can upgrade between branches (1.0 to 1.1) through the backend. 

> I use a few of the rainlab-plugins and I've seen that you were successful in forking these as well – will new plugin updates at one point automatically come from the Winter-forks?

We will not automatically replace RainLab plugins with Winter plugins, you'll have to do that manually by removing the rainlab ones from the filesystem (not through the backend, as that will also remove all their data) and then installing the winter ones.

> Which files are actually NOT overwritten by the updater? (Just asking because I'm really angry at OCMS for closing the source and want to purge that name from all files)

If you want to do this, the best option would be to just create a new winter project with composer, and then copy over your `plugins/`, `themes/`, & `storage/` directories as well as your `.env` file if you use one and the options you've customized in your `config/` directory. This way you'll also get immediate access to the latest winter features.

> Is there anything I should keep out to watch which might break once my pages depart completely from the October branches?

There shouldn't be anything that would break, at this point your site is more likely to break upgrading to October 2.0 than it is migrating to Winter. If anything does happen though let us know here and we'll most likely be able to solve it.

> P.S.: Prepare for an incomping sponsorship.

Awesome, thanks for the support!

**Reply by nathanlesage** [reply]
Perfect, thank you! :)
